src/main/api/opds.ts

```diff
@@ -66,14 +60,22 @@ export class OpdsApi {
 
     public async browse(data: any): Promise<THttpGetOpdsResultView> {
         const { url } = data;
-        return await httpGet(url, {}, async (opdsFeedData) => {
+        return await httpGet(url, {
+            timeout: 10000,
+        }, async (opdsFeedData) => {
             // let opds2Publication: OPDSPublication = null;
             let opds2Feed: OPDSFeed = null;
 
             if (opdsFeedData.isFailure) {
                 return opdsFeedData;
             }
 
+            debug("opdsFeed content-type", opdsFeedData.contentType);
+            if (!opdsFeedData.contentType.startsWith("application/json") &&
+                !opdsFeedData.contentType.startsWith("application/atom+xml")) {
+                throw new Error("bad content-type");
```

Comment:
This will be hard to debug at runtime (especially when troubleshooting remotely, when people send bug reports), so I suggest including at least the URL and the HTTP Content-Type, such as:
`throw new Error(`Not a valid OPDS HTTP Content-Type: ${opdsFeedData.contentType.url} => ${opdsFeedData.contentType}`);`

Reply:
done

Reply:
https://github.com/readium/readium-desktop/pull/542/commits/a5d70e3c7e09d8d5a0e9810224444aed7083095c